### DISCUSSION TITLE
Playtime: Follow actions

### DISCUSSION THREAD

**Discussion by helgoboss** [helgoboss/helgobox#1044] [Ideas]

**Comment by ThoughtPhotography** [comment]
**Follow actions** https://www.ableton.com/en/live-manual/12/launching-clips/#follow-actions for clips (slots), scenes (rows), and tracks (columns). Aside from experimentation/probability they're also widely used for simplifying the number of clips required for a live performance. 

For example, the multiplier sets the number of loops before a follow action is triggered. So if set to _4x_ and _next_ a 4 bar loop would play for 16 bars until moving to the next clip, or you could have a 1 bar loop with _8x_ to if you want it to work well in a row with 8 bar loops etc.

Track follow actions act as a global override for all clips in a track, but would also be useful as the _default_ for all clips that haven't been changed. That way you could easily change a whole column to play in reverse order (previous clip), but also have a clip with a random follow action which will jump to a different clip in the track and continue playing in reverse order from there etc. All you need to do is change the track follow back to next and normal order is restored. Without track follow actions every clip in a track would need changing manually to try something simple, then change them all back again (potentially dozens of steps vs. 2)

![follow actions](https://github.com/user-attachments/assets/da53cad0-cce8-4521-b826-90e3e0ff17e1)

https://forum.cockos.com/showpost.php?p=2769543&postcount=365

**Reply by Pav9146** [reply]
Эта функция очень нужна. Если это заработает, то это будет просто потрясающе!

**Reply by helgoboss** [reply]
Would appreciate if you could use English, thanks.

You can vote for this by clicking the arrow on the top.

**Reply by Pav9146** [reply]
This feature is very necessary. If it works, then it will be just amazing

**Reply by urtaur** [reply]
It would be amazing

**Reply by sevenseas298** [reply]
A great idea!

Implementation of a text-based clip/scene sequencer for Playtime 2 might be worth considering.

I use ComputerScare's "I Love Cookies" VCV Rack module in many of my Playtime 2 clip sequencing patches.

...simple syntax and very useful - powerful randomization and sequence variation capabilities.

**Reply by PianoGeek15** [reply]
100% support this idea! 

It would also be interesting to implement some sort of "Clip Lineup" where you can create a long arrangement using just a few rows of clips. Use case scenario would be using Playtime to run tracks for a live event. The advantage to the Clip Lineup would be that you could have just one row for each section of the song and bounce between them, instead of creating a long stack of sections (and I'd rather not have to manually trigger song sections live).

**Reply by helgoboss** [reply]
@PianoGeek15 When I hear "Clip lineup", I immediately think of this existing feature: https://docs.helgoboss.org/playtime/advanced-usage-scenarios/control-with-items.html
